File: app/src/main/java/im/threads/android/data/Card.java
package im.threads.android.data;

import java.io.Serializable;

import androidx.core.util.ObjectsCompat;

public class Card implements Serializable {
    private final String userId;
    private final String clientData;
    private final String appMarker;
    private final String clientIdSignature;
    private final String authToken;
    private final String authSchema;

    public Card(final String userId, String clientData, final String appMarker, final String clientIdSignature, final String authToken, final String authSchema) {
        this.userId = userId;
        this.clientData = clientData;
        this.appMarker = appMarker;
        this.clientIdSignature = clientIdSignature;
        this.authToken = authToken;
        this.authSchema = authSchema;
    }

    public String getUserId() {
        return userId;
    }

    public String getClientData() {
        return clientData;
    }

    public String getAppMarker() {
        return appMarker;
    }

    public String getClientIdSignature() {
        return clientIdSignature;
    }

    public String getAuthToken() {
        return authToken;
    }

    public String getAuthSchema() {
        return authSchema;
    }

    @Override
    public boolean equals(final Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        final Card card = (Card) o;
        return ObjectsCompat.equals(userId, card.userId);
    }

    @Override
    public int hashCode() {
        return userId != null ? userId.hashCode() : 0;
    }
}

File: app/src/main/java/im/threads/android/data/Card.kt
package im.threads.android.data

import android.os.Parcelable
import androidx.core.util.ObjectsCompat
import kotlinx.parcelize.Parcelize

@Parcelize
data class Card(
    val userId: String,
    val clientData: String,
    val appMarker: String,
    val clientIdSignature: String,
    val authToken: String,
    val authSchema: String
) : Parcelable {
    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }
        if (other == null || javaClass != other.javaClass) {
            return false
        }
        val card = other as Card
        return ObjectsCompat.equals(userId, card.userId)
    }

    override fun hashCode(): Int {
        return userId.hashCode()
    }
}
